schedulearn-frontend/src/components/Contexts/UsernameContext.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Component } from "react";
+
+interface UsernameProps {
+  children: React.ReactNode;
+}
+
+export interface UsernameState {
+  username: string;
+  setUsername: (username: string) => void;
+}
+
+export const UsernameContext = React.createContext<UsernameState>({
+  username: "not-from-state", 
+  setUsername: (username: string) => {console.log("Username context has not been set yet!", username);},
+});
+
+export class UsernameProvider extends Component<UsernameProps> {
+  state = {
+    username: "not-yet-set",
+  }
+
+  setUsername = (username: string): void => {
+    this.setState({
```

Comment:
We need to somehow save more than just the username into the context (for instance we might need UserId for querying certain data, such as calendar events belonging to the user).

I suggest you create an interface User (based on backend User entity) and save the whole object instead of just a username.

Reply:
Replace with User class